### PR TITLE
Enables support to specify the region.

### PR DESCRIPTION
**Pull request by marcgascon**
By default the gem fog-aws is using the region 'us-east-1' as we could see here:
https://github.com/fog/fog-aws/blob/master/lib/fog/aws/compute.rb#L302

If I specify the endpoint: https://ec2.eu-west-1.amazonaws.com, (the region is 'us-east-1', as we have seen) then the connection returns OK but when I try to list the route table ruby throws an exception because the endpoint and region are not matching.

If the user didn't specify the region I will let him know that by default he is using 'us-east-1'

Comments are more than welcome.
Thanks.

### CODE REVIEW

**Comment by jeffbyrnes** [comment]
@marcgascon thanks for the contribution! I’ll let @eherot chime in here on this, as I haven‘t mucked about with this repo much at all.

**Comment by eherot** [comment]
In principle I like the idea of adding region support but I would rather not break backwards compatibility. How about instead we do something like:
```
options[:region] = @conf['region'] if @conf['region']
```
That way people can specify the region if they want but otherwise it defaults to `us-east-1` (which would mimic most other Amazon services.

**Comment by marcgascon** [comment]
Ok @eherot it sounds good, but then I would like to update the Readme to clarify some things and to make easier to use this gem.

I will do another commit doing these changes and I will let you know.

**Comment by marcgascon** [comment]
New commit with your changes proposal and update of the documentation is added.

Thanks for the comments. It seems that the gem could be very useful for me. Can you please review it? Also, if there is something more that I could do please tell me.

Thanks!

File: README.md [hunk 1]
```diff
@@ -39,13 +39,23 @@ nodes:
   i-00000003: 10.0.2.1
 ```
 
-Optional properties include:
+Optional properties include (If you don't specify these variables, the following values are the default ones):
```
Comment (eherot) [inline README.md]: I think for brevity's sake you could just make this say "Values shown are the defaults"

File: README.md [hunk 2]
```diff
@@ -39,13 +39,23 @@ nodes:
...
 heartbeat_interval: 10
 ```
 
+Optional aws configuration include:
+```yaml
+aws_access_key_id: YOUR ACCESS KEY
+aws_secret_access_key: YOUR SECRET KEY
+region: us-east-1
+```
+Note that:
```
Comment (eherot) [inline README.md]: And then here you could drop the note about `us-east-1`

File: README.md [hunk 3]
```diff
@@ -39,13 +39,23 @@ nodes:
...
 
-Optional properties include:
+Optional properties include (If you don't specify these variables, the following values are the default ones):
 ```yaml
 pings: 3
 ping_timeout: 1
 heartbeat_interval: 10
 ```
 
+Optional aws configuration include:
```
Comment (eherot) [inline README.md]: Oh and AWS should be capitalized.

Comment (eherot) [inline README.md]: Actually, on second thought, why do you feel that we need to have the AWS options as part of a separate block from the other optional configuration?

Comment (marcgascon) [inline README.md]: Because I think that it's clearest and more readable to sepparate in blocks the config file.

**Comment by marcgascon** [comment]
Ok I will do all the changes you are proposing.

**Comment by jeffbyrnes** [comment]
By the way, kyle-house-sp/nat-monitor@4dfeaa75030b317b2820b4ba76aece93b17d9da8 seems to have gone a similar route (cc @kyle-house-sp). I wager they might be interested to see this feature land here.

**Comment by kyle-house-sp** [comment]
sorry about that, I should've pushed my PR back up...happy to do so, but this looks like it's on the same track!

**Comment by marcgascon** [comment]
Hello @eherot PR is updated with your corrections. How do you see it now?

@jeffbyrnes I was sawing the @kyle-house-sp fork and I found very interesting his first commit allowing to work with only 2 instances in the quorum. And about the region I think that we can merge this PR and he can do a rebase to continue working with the first commit. What do you think?

For the future work @eherot it could be very useful if you can describe an example (in the README) of the minimum AWS set up (Instances, AZ's, etc). I am experiencing a lot of troubleshooting using the gem.

Thanks for all your comments guys.

Appreciated!

**Comment by marcgascon** [comment]
Thanks for merging @eherot ! When do you think that I will be able to download this new version from rubygems?

And @jeffbyrnes, yes it could be very useful a basic diagram or explanation of the use case to see how many instances we need to use this gem and the communication between them because from now on I wasn't able to work with the gem as expected, but I will keep investigating.

Thank you all!

**Comment by eherot** [comment]
@marcgascon sorry it took so long. The new release (v1.0.11) should be available on RubyGems now.

**Pull request merged**